hist/hist/src/THnBase.cxx

```diff
@@ -195,13 +197,20 @@ void THnBase::Init(const char* name, const char* title,
                    Int_t chunkSize /*= 1024 * 16*/)
 {
    SetNameTitle(name, title);
-
+   if (!axes) {
+      ::Error("THnBase::Init", "axes is null");
+      return;
+   }
    TIter iAxis(axes);
    const TAxis* axis = nullptr;
    Int_t pos = 0;
    Int_t *nbins = new Int_t[axes->GetEntriesFast()];
    while ((axis = (TAxis*)iAxis())) {
       TAxis* reqaxis = new TAxis(*axis);
+      if (!reqaxis) {
+         ::Error("THnBase::Init", "reqaxis is null");
+         continue;
+      }
```

Comment:
Can this condition even happen? As long as `axis!=nullptr`, then the copy-constructor will return a proper `Taxis` object. If `axis==nullptr`, then line 209 will actually crash. Probably the check should be `if (!axis)`. At the same time, this would mean that the `TObjArray` is holding some empty data, which seems weird, but still.

In that case the error should be something like `Axis could not be found, cannot create new one` or something similar

Reply:
You're right!